JenifarFathimaH/00 - Projects/04 - Frontend Projects/E-Commerce Project/Assets/JS/Admin/AddProduct.js

```diff
@@ -0,0 +1,99 @@
+const PRODUCT_NAME = document.getElementById("inp-productName");
+const PRODUCT_ID = document.getElementById("inp-productId");
+const PRODUCT_PRICE = document.getElementById("inp-productPrice");
+const PRODUCT_DESCRIPTION = document.getElementById("inp-productDescription");
+const ALERT_MESSEGE = document.querySelector('.alert_messege');
+
+const SUBMIT_BUTTON = document.getElementById("inp-SubmitButton");
+const UPDATE_BUTTON = document.getElementById("inp-UpdateButton");
+
+const EDIT_INDEX = parseInt(localStorage.getItem("editIndex"));
+
+const PRODUCT_DETAILS = JSON.parse(localStorage.getItem("productDetails")) || [];
+
+let AddProductMethods = {};
+let EditAndUpdateMethods = {};
+
+AddProductMethods.FillProductDetails = function () {
+    let strProductName = PRODUCT_NAME.value;
+    let nProductId = PRODUCT_ID.value;
+    let nProductPrice = PRODUCT_PRICE.value;
+    let strProductDescription = PRODUCT_DESCRIPTION.value;
+
+    if (!strProductName || !nProductId || !nProductPrice || !strProductDescription) {
+        ALERT_MESSEGE.innerHTML = `<p>All fields are Required<p>`;
+        return;
+    }
+    AddProductMethods.SaveProduct();
+
+}
+
+AddProductMethods.SaveProduct = function () {
+    let productDetailObject = {
+        strProductName: PRODUCT_NAME.value,
+        nProductId: PRODUCT_ID.value,
+        nProductPrice: PRODUCT_PRICE.value,
+        strProductDescription: PRODUCT_DESCRIPTION.value
+    }
+    PRODUCT_DETAILS.push(productDetailObject);
+
+    localStorage.setItem("productDetails", JSON.stringify(PRODUCT_DETAILS));
+    let url = window.location.href.replace("AddProduct.html","ViewEditProduct.html");
+    window.location.href = url;
+    return false;
+}
+
+
+EditAndUpdateMethods.EditProductDetails = function(){
+
+    UPDATE_BUTTON.type = "button";
+
+    let strProductName = PRODUCT_NAME.value;
+    let nProductId = PRODUCT_ID.value;
+    let nProductPrice = PRODUCT_PRICE.value;
+    let strProductDescription = PRODUCT_DESCRIPTION.value;
+
+    let item = PRODUCT_DETAILS[EDIT_INDEX];
+
+    if (item) {
+        item.strProductName = strProductName;
+        item.nProductId = nProductId;
+        item.nProductPrice = nProductPrice;
+        item.strProductDescription = strProductDescription;
+
+        localStorage.setItem("productDetails", JSON.stringify(PRODUCT_DETAILS));
+        let url = window.location.href.replace("AddProduct.html", "ViewEditProduct.html");
+        window.location.href = url;
+
+    }
+    else {
+        console.error("Item not found for editing!");
+    }
+
+}
+
+document.addEventListener("DOMContentLoaded", function () {
```

Comment:
This should be placed at the top after the global variable declaration... (Not a mandatory change. Just a recommendation)

Reply:
Done